### DISCUSSION TITLE
What happens to previous command-line output?

### DISCUSSION THREAD

**Discussion by photismos** [leo-arch/clifm#290] [General]
Hi. One of the features I love is how clean clifm keeps the terminal. I simply hit Enter and I get a clean slate. But every once in a while, I wish I had some command-line output back so I might document it for later reference. Is there any way to get that output back, or I better have the presence of mind to grab it before I hit Enter? If not, well, whatever. It's not going to stop me from learning more clifm, thank you very much!

**Comment by leo-arch** [comment]
Hi @photismos. Well, **clifm** does not keep any screen buffer, and this includes previous outputs. This is a task performed rather by the terminal emulator itself: what **clifm** does is just to send an escape code (`\e[H\e[2J\e[3J`) to the terminal, which interprets this code and clears the screen. Now, most modern terminal emulators really flush the screen buffer (that is, no previous state is kept), while some others only scrolls the terminal down, thus keeping previous outputs (you only need to scroll up to get it back).

You can check it out yourself by just sending this escape code directly to your terminal:

```sh
echo -e "\e[H\e[2J\e[3J"
```

I guess this behavior might be controlled via some terminal setting, but it depends on the terminal you're using.
